lib/spack/spack/installer.py

```diff
@@ -998,6 +1000,33 @@ def __init__(
         self.attempts = 0
         self._update()
 
+        # Is this task to install a compiler
+        self.compiler = compiler
+
+        # Handle bootstrapped compiler
+        #
+        # The bootstrapped compiler is not a dependency in the spec, but it is
+        # a dependency of the build task. Here we add it to self.dependencies
+        if compiler:
+            compiler_spec = self.pkg.spec.compiler
+            arch_spec = self.pkg.spec.architecture
+            strict = spack.concretize.Concretizer().check_for_compiler_existence
+            if (
+                not spack.compilers.compilers_for_spec(compiler_spec, arch_spec=arch_spec)
+                and not strict
+            ):
+                # The compiler is in the queue, identify it as dependency
+                dep = spack.compilers.pkg_spec_for_compiler(compiler_spec)
+                dep.constrain(f"platform={str(arch_spec.platform)}")
+                dep.constrain(f"os={str(arch_spec.os)}")
+                dep.constrain(f"target={arch_spec.target.microarchitecture.family.name}:")
+                dep.concretize()
+                dep_id = package_id(dep.package.spec)
+                self.dependencies.add(dep_id)
```

Comment:
No, it will not be needed if this goes in after the removal of bootstrapping compilers. I think it will be easier if we merge this first though (as the refactoring involved in this is harder to rebase) and then also rip this out as part of that PR. But we should talk through that ordering problem with @alalazo

Reply:
I'm fine with any ordering, but would like to get #46237 in soon (to reduce the size of #45189). I think, if I leave the cleanup of `*Task` attributes to this PR there won't be much overlap. See https://github.com/spack/spack/pull/46237#issuecomment-2333321552